src/borg/remote.py

```diff
@@ -176,7 +177,17 @@ def __init__(self, location, create=False, lock_wait=None, lock=True, args=None)
         if version != RPC_PROTOCOL_VERSION:
             raise Exception('Server insisted on using unsupported protocol version %d' % version)
         try:
-            self.id = self.call('open', self.location.path, create, lock_wait, lock)
+            # Because of protocol versions, only send append_only if necessary
+            if append_only:
+                try:
+                    self.id = self.call('open', self.location.path, create, lock_wait, lock, append_only)
+                except self.RPCError as err:
+                    if err.rootCause == 'TypeError':
+                        raise Exception('Server does not support --append-only') from err
+                    else:
+                        raise err
```

Comment:
Just `raise` is enough - it will re-raise the currently handled exception (here in `err`) if no exception is specified. You can also leave the else: away if you like, since the if: block raises, when executed.